# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1497744486 14400
#      Sat Jun 17 20:08:06 2017 -0400
# Node ID b5a9cd6de1b8f18ecc1af1c24f74530ecafc9442
# Parent  ff68c9d024b84e6542467b9bb8cdcd99ecf0a549
avoid possible memory error in parser

* oct-parse.in.yy (opt_fcn_list): For empty list case, return
allocated emtpy tree_statement_list instead of 0.

diff --git a/libinterp/parse-tree/oct-parse.in.yy b/libinterp/parse-tree/oct-parse.in.yy
--- a/libinterp/parse-tree/oct-parse.in.yy
+++ b/libinterp/parse-tree/oct-parse.in.yy
@@ -431,17 +431,17 @@ list            : list1 opt_sep
 
 list1           : statement
                   { $$ = parser.make_statement_list ($1); }
                 | list1 sep statement
                   { $$ = parser.append_statement_list ($1, $2, $3, true); }
                 ;
 
 opt_fcn_list    : // empty
-                  { $$ = 0; }
+                  { $$ = new octave::tree_statement_list (); }
                 | fcn_list
                   { $$ = $1; }
                 ;
 
 fcn_list        : fcn_list1 opt_sep
                   {
                     YYUSE ($2);
 
